allure-generator/src/main/java/io/qameta/allure/allure2/Allure2ResultsReader.java

```diff
@@ -75,7 +77,12 @@ public Allure2ResultsReader(AttachmentsStorage storage) {
                     TestCaseResult dest = new TestCaseResult();
                     dest.setUid(generateUid());
 
-                    dest.setId(result.getId());
+                    String id = Optional.ofNullable(result.getId()).orElseGet(() -> {
```

Comment:
> We discussed that we wand to generate an id if it has not been provided

Nope, that is gonna be a feature - use null id to disable history & statistics for test case